app/controllers/idv/in_person/state_id_controller.rb

```diff
@@ -16,6 +16,52 @@ def show
         render :show, locals: extra_view_variables
       end
 
+      def update
+        # don't clear the ssn when updating address, clear after SsnController
+        clear_future_steps_from!(controller: Idv::InPerson::SsnController)
+
+        pii_from_user = flow_session[:pii_from_user]
+        initial_state_of_same_address_as_id = flow_session[:pii_from_user][:same_address_as_id]
```

Comment:
Why are we grabbing from the flow_session here and not from the pii_from_user var you have on line 23? Can they become out of sync?

Reply:
Also it seems kinda random where we grab directly from the flow_session and the pii_from_user var. Is there reasoning behind it? I might just not be able to grok it.

Reply:
good point, i updated to use `pii_from_user` throughout the file